Bundling viewer

stevejpurves opened the pull request:
purpose of this PR is to attempt to bundle all itk viewer deps (without the WASM) into a single file (ESM) that can then be dynamically loaded in the NodeRenderer

stevejpurves committed:
🚧 wip

stevejpurves committed:
📦🔒

stevejpurves commented:
@PaulHax no success yet but wanted to share the approach I was working on

stevejpurves committed:
consume viewer async

stevejpurves commented:
currrently hitting this

<img width="969" alt="renderer tsx — Untitled (Workspace) 2024-07-12 15-35-30" src="https://github.com/user-attachments/assets/fea4e067-a31b-4486-8e02-ea9fc615d303">

PaulHax commented:
Guessing that the build is pulling from this file:
https://github.com/InsightSoftwareConsortium/ITK-Wasm/blob/main/packages/core/typescript/itk-wasm/src/index-node.ts

But we want tsup to get the browser conditional index.js
https://github.com/InsightSoftwareConsortium/ITK-Wasm/blob/main/packages/core/typescript/itk-wasm/package.json#L12-L14

Is it possible to feed that `-conditions` flag to tsup or the build tool?
https://nodejs.org/api/packages.html#resolving-user-conditions

@thewtex

thewtex committed:
Use --platform browser in tsup

This ensures we get the correct package.json `exports` conditions.

Which is passed to esbuild

  https://esbuild.github.io/api/#platform

to get the correct conditions for the `browser`:

  https://nodejs.org/api/packages.html#resolving-user-conditions

The default for tsup is `node`.

thewtex commented:
PR against this branch to address the build here: https://github.com/curvenote/itk-plugins/pull/5 🎇

stevejpurves committed:
Merge pull request #5 from thewtex/tsup-platform

Use --platform browser in tsup

stevejpurves commented:
@PaulHax @thewtex I think we will be integrating this, which might be a generic vehicle to get any renderer into myst https://github.com/manzt/anymyst

thewtex commented:
@stevejpurves awesome! We will look into that soon as we explore re-adding ipywidget support to [itkwidgets](https://github.com/InsightSoftwareConsortium/itkwidgets)

At the same time, the itk-viewer plugin will also be helpful because it will not require writing or running any code.
